src/Tasks/Microsoft.NET.Build.Tasks/SingleProjectInfo.cs

```diff
@@ -34,12 +42,58 @@ public static SingleProjectInfo Create(string projectPath, string name, string v
                 resourceAssemblies.Add(new ResourceAssemblyInfo(culture, relativePath));
             }
 
-            return new SingleProjectInfo(projectPath, name, version, resourceAssemblies);
+            return new SingleProjectInfo(projectPath, name, version, $"{name}.dll", resourceAssemblies);
         }
 
-        public string GetOutputName()
+        public static Dictionary<string, SingleProjectInfo> CreateFromProjectReferences(
+            ITaskItem[] projectReferencePaths,
+            ITaskItem[] projectReferenceSatellitePaths)
         {
-            return $"{Name}.dll";
+            Dictionary<string, SingleProjectInfo> projectReferences = new Dictionary<string, SingleProjectInfo>();
+
+            foreach (ITaskItem projectReferencePath in projectReferencePaths)
+            {
+                string sourceProjectFile = projectReferencePath.GetMetadata("MSBuildSourceProjectFile");
+
+                if (string.IsNullOrEmpty(sourceProjectFile))
+                {
+                    throw new Exception($"Could not find valid 'MSBuildSourceProjectFile' metadata on ReferencePath '{projectReferencePath.ItemSpec}'");
```

Comment:
Should we do something else here?  Like skip it?  It feels like that will just cause other issues downstream...

Maybe log a warning?